samza-sql-shell/src/main/java/org/apache/samza/sql/client/interfaces/CommandHandler.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.sql.client.interfaces;
+
+import java.io.PrintWriter;
+import org.apache.samza.sql.client.cli.CliCommand;
+import org.apache.samza.sql.client.cli.CliEnvironment;
+import org.apache.samza.sql.client.cli.CliShell;
+import org.jline.terminal.Terminal;
+
+
+public interface CommandHandler {
+  void initCommandHandler(CliShell shell, CliEnvironment env, PrintWriter printWriter, Terminal terminal,
```

Comment:
We can get the `printWriter` from `terminal.writer()`, so we can save a parameter here.

Also, consider remove the `shell` parameter. We may not want a `commandhandler` to access the Shell class directly. Encapsulate as much as we can, after all. I don't see code in the `commandhandler` implementation use it, but please correct me if I am mistaken.

Reply:
removed printWriter, but need shell (used in commandSelect > view.open(shell, queryResult) at line 437), also encapsulation is preserved here since we only pass the shell to CliView...

Reply:
Yeah... Thanks. I'll clean up CliView later. CliView actually only uses the terminal, executor and context.

Reply:
Just one trivia: `init` shall be sufficient, instead of `initCommandHandler`?

Reply:
done